languages/go/menhir/parser_go.mly

```diff
@@ -1021,7 +1023,25 @@ fnliteral: fnlitdcl lbrace listsc(stmt) "}"
 fnlitdcl: fntype { $1 }
 
 arg_type:
-|       name_or_type { ParamClassic { pname= None; ptype = $1; pdots = None } }
+|       name_or_type {
+    (match $1 with
+    (* This means a param of the form $...<ID>.
+     * Ordinarily, in Golang, a singular identifier is interpreted as an
+     * anonymous argument of the type of that identifier.
+     * So func foo(int) is allowed.
+
+     * When we see a metavariable ellipsis, it's a little silly to consider that
+     * as a type. If someone writes func foo ($...ARGS), they probably just meant to
+     * capture all of the args of the function, and not an anonymous argument of type $...ARGS.
+
+     * So we parse it as such here.
+     *)
+    | TName [ (s, _) as id ] when AST_generic.is_metavar_ellipsis s ->
```

Comment:
You can also guard it with Flag.sgrep_guard, which make sure we are in parsing pattern mode.